pkg/cmd/server/server.go

```diff
@@ -703,21 +704,22 @@ func (s *server) runControllers(defaultVolumeSnapshotLocations map[string]string
 	restoreControllerRunInfo := func() controllerRunInfo {
 		restorer, err := restore.NewKubernetesRestorer(
 			s.veleroClient.VeleroV1(),
-			s.discoveryHelper,
-			client.NewDynamicFactory(s.dynamicClient),
+			s.mgr.GetClient(),
 			s.config.restoreResourcePriorities,
-			s.kubeClient.CoreV1().Namespaces(),
-			s.resticManager,
+			// Because we don't want any restic features for our use case.
```

Comment:
same as above. What would happen if we just hand over this stuff? We shouldn't use it anyway? (same below with the podexec)